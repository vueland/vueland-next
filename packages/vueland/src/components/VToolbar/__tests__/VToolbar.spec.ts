import { mount } from '@vue/test-utils'
import { h } from 'vue'
import { VToolbar, VToolbarNavBtn, VToolbarLogo, VToolbarItems, VToolbarContent } from '../index'

describe('VToolbar', () => {
  const mountFunction = (options = {}) => mount(VToolbar, {
    ...options,
    global: {
      provide: {
        $v: {},
      },
    },
    slots: {
      default: () => h(VToolbarContent, {},
        {
          default: () => [ h(VToolbarNavBtn), h(VToolbarLogo), h(VToolbarItems) ]
        }
      )
    }
  })

  it('should mount component and match snapshot', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
