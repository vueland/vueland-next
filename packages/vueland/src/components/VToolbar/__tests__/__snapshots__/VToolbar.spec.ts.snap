// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VToolbar should mount component and match snapshot 1`] = `
<div class="v-toolbar v-toolbar--base"
     style="height: 56px;"
>
  <div class="v-toolbar__content">
    <div class="v-toolbar__nav-btn">
      <button class="v-icon v-icon--clickable fas fa-bars sm">
      </button>
    </div>
    <div class="v-toolbar__logo">
    </div>
    <div class="v-toolbar__items">
    </div>
  </div>
</div>
`;
